File: Pro_ODP_NET_for_Oracle_Database_11g-4541/c_sharp/Chapter 4/InsertCLOBData.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Oracle.DataAccess.Client;
using Oracle.DataAccess.Types;

namespace Chapter_4
{
    public partial class InsertCLOBData : Form
    {
        public InsertCLOBData()
        {
            InitializeComponent();
        }

        private void button2_Click(object sender, EventArgs e)
        {
            string _connstring = "Data Source=localhost/NEWDB;User Id=EDZEHOO;Password=PASS123;";
            int _recordsAffected;
            try
            {
                OracleConnection _connObj = new OracleConnection(_connstring);
                _connObj.Open();
                OracleCommand _cmdObj = _connObj.CreateCommand();
                _cmdObj.CommandText = "INSERT INTO ProductFiles(ProductID, Remarks) VALUES(:ProductID,:Remarks)";
                _cmdObj.Parameters.Add(new OracleParameter("ProductID", txtProductID.Text));
                OracleClob _clobObj = new OracleClob(_connObj);
                _clobObj.Write(txtRemarks.Text.ToCharArray(), 0, txtRemarks.Text.Length);
                _cmdObj.Parameters.Add(new OracleParameter("Remarks", _clobObj));
                _recordsAffected = _cmdObj.ExecuteNonQuery();
                if (_recordsAffected == 1) { MessageBox.Show("CLOB saved!"); }
                _connObj.Close();
                _connObj.Dispose();
                _connObj = null;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
        }
    }
}

File: Pro_ODP_NET_for_Oracle_Database_11g-4541/vb_dot_net/Chapter4VBNET/InsertCLOBData.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Oracle.DataAccess.Client
Imports Oracle.DataAccess.Types


Namespace Chapter_4
End Namespace 'Chapter_4
 _
Class InsertCLOBData
    Inherits Form
    '
    'ToDo: Error processing original source shown below
    '
    '
    '------------^--- 'class', 'struct', 'interface' or 'delegate' expected
    '
    'ToDo: Error processing original source shown below
    '
    '
    '--------------------^--- Syntax error: ';' expected
    Public Sub New()
        InitializeComponent()
    End Sub 'New


    Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
        Dim _connstring As String = "Data Source=localhost/NEWDB;User Id=EDZEHOO;Password=PASS123;"
        Dim _recordsAffected As Integer
        Try
            Dim _connObj As New OracleConnection(_connstring)
            _connObj.Open()
            Dim _cmdObj As OracleCommand = _connObj.CreateCommand()
            _cmdObj.CommandText = "INSERT INTO ProductFiles(ProductID, Remarks) VALUES(:ProductID,:Remarks)"
            _cmdObj.Parameters.Add(New OracleParameter("ProductID", txtProductID.Text))
            Dim _clobObj As New OracleClob(_connObj)
            _clobObj.Write(txtRemarks.Text.ToCharArray(), 0, txtRemarks.Text.Length)
            _cmdObj.Parameters.Add(New OracleParameter("Remarks", _clobObj))
            _recordsAffected = _cmdObj.ExecuteNonQuery()
            If _recordsAffected = 1 Then
                MessageBox.Show("CLOB saved!")
            End If
            _connObj.Close()
            _connObj.Dispose()
            _connObj = Nothing
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try
    End Sub 'button2_Click
End Class 'InsertCLOBData
'
'ToDo: Error processing original source shown below
'
'
'-^--- expression expected